framework/base/Security.php

```diff
@@ -455,6 +459,15 @@ public function generateRandomKey($length = 32)
 
         $bytes = '';
 
+        if (function_exists('random_bytes')) {
+            try {
+                $bytes .= random_bytes($length);
```

Comment:
@cebe 

1st point - agreed.

2nd point. It's probably historical. In the [original version](http://www.yiiframework.com/extension/randomness/) of this method, lots of sources were tried. Nesting everything was a mess, iirc, and appending was tidier. It was also trying to accommodate the fact that some sources could return a value that wasn't as long as requested.

Your comment makes me think I should reconsider that structure. But first it would be good to get some more opinion on #9879, in particular, is `if (function_exists('mcrypt_create_iv')) {...}` in our out?